modules/@apostrophecms/rich-text-widget/ui/apos/components/AposTiptapLink.vue

```diff
@@ -114,10 +106,10 @@ export default {
   },
   computed: {
     buttonActive() {
-      return !isEmpty(this.value.data);
+      return this.value.data && this.value.data.href;
     },
     hasSelection() {
-      return this.editor.selection.from !== this.editor.selection.to;
+      return this.editor.view.state.selection.ranges[0].$from.pos !== this.editor.view.state.selection.ranges[0].$to.pos;
```

Comment:
Will ranges[0] always exist?

Reply:
I think they do, but I have updated this to use the `textBetween` tiptap helper as a safer bet